common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -279,6 +279,74 @@ def verify_current_content_visibility(cls, date, hide_after_date):
             datetime.now(UTC) < date
         )
 
+    def _has_access_check(self, *args):
```

Comment:
Would it be possible to mock out the return value for the `has_access` function to return a specific value based off the parameters? Or do some sort of mock like in https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/tests/test_sequence.py#L308-L316?

Reply:
I did this because i think has_access was being called multiple times in the stack trace and because I was importing it inside a function to avoid a circular import, i couldn't override only the has_access calls inside this file, so created this helper method for the test 
happy to mock out has_access directly if we can figure out a way to do that

Reply:
Not sure how applicable these are, but found a couple of places that mocks has_access in existing tests:
https://github.com/edx/edx-platform/blob/master/lms/djangoapps/edxnotes/tests.py#L659-L674
https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/tests/test_module_render.py#L1732

Reply:
you'll notice those mocks reference the relevant files, rather than access.py where the function is defined  
this means that calling has_access from those files will be mocked, but other calls will not be mocked 
the issue here (if i understand it correctly), is that because i'm importing has_access inside the function and not at the top, i can't mock has_access inside only this module because it isn't available at the module level (which means other calls to has_access from other places, which are frequent, will get mocked as well) 
not sure if that helps clarify my above comment